### PR TITLE
[test] Skip Safari and Firefox on broken tests

### PR DESCRIPTION
**Pull request by alexfauquette**
From https://github.com/mui/mui-x/pull/4645#issuecomment-1132887003

Skip tests on Firefox and Safari because they fail without apparent reason

### CODE REVIEW

File: packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.test.tsx [hunk 1]
```diff
@@ -46,7 +46,10 @@ describe('<StaticTimePicker />', () => {
     }),
   );
 
-  it('should allows view modification, but not update value when `readOnly` prop is passed', () => {
+  it('should allows view modification, but not update value when `readOnly` prop is passed', function test() {
+    if (/safari/i.test(window.navigator.userAgent) || /firefox/i.test(window.navigator.userAgent)) {
+      this.skip();
```
Comment (oliviertassinari) [inline packages/x-date-pickers/src/StaticTimePicker/StaticTimePicker.test.tsx:51]: For each test, could we leave a comment on what's missing to make it work? I believe we have done this systematically in the past. 

For instance:
```suggestion
      // Only run in supported browsers
      if (typeof Touch === 'undefined') {
        this.skip();
      }
```

File: packages/x-date-pickers/src/ClockPicker/ClockPicker.test.tsx [hunk 2]
```diff
@@ -194,7 +194,10 @@ describe('<ClockPicker />', () => {
     expect(Math.max(...hours)).to.equal(23);
   });
 
-  it('should display options, but not update value when readOnly prop is passed', () => {
+  it('should display options, but not update value when readOnly prop is passed', function test() {
+    if (/safari/i.test(window.navigator.userAgent) || /firefox/i.test(window.navigator.userAgent)) {
+      this.skip();
+    }
```
Comment (oliviertassinari) [inline packages/x-date-pickers/src/ClockPicker/ClockPicker.test.tsx:200]: For each test, could we leave a comment on what's missing to make it work? I believe we have done this systematically in the past. 

For instance:
```suggestion
    // Only run in supported browsers
    if (typeof Touch === 'undefined') {
      this.skip();
    }
```

Comment (alexfauquette) [reply packages/x-date-pickers/src/ClockPicker/ClockPicker.test.tsx]: 👍 Sounds easier to read

Comment (oliviertassinari) [reply packages/x-date-pickers/src/ClockPicker/ClockPicker.test.tsx]: And more reliable, if tomorrow the API is available, then the test starts running.

**Pull request merged**